### ISSUE TITLE
Print test environment information

### ISSUE THREAD

**Issue by lann** [fermyon/conformance-tests#15]
It would be nice to have information about the inputs in test output:

- build information for the crates in this repo
- (ideally) build information for the runtime
- any relevant configuration for the test run

**Comment by rylev** [comment]
I'm not sure I understand what you're asking for. Can you provide some example output you would expect to see?

**Comment by lann** [comment]
I'm mostly thinking of the types of information you get from e.g. [`vergen`](https://docs.rs/vergen/8.3.1/vergen/#use-cases)

```
Runtime:            spin 3.0.0
conformance-tests:  0.1.0-9-g46f83e1
test-environment:   0.1.0-9-g46f83e1
rustc:              1.52.0-nightly
Target:             x86_64-unknown-linux-musl
```

**Comment by rylev** [comment]
Sounds good. One issue is that only some of this information is known to the conformance tests themselves. The runtime under test for example is not known to the conformance tests and will have to be provided by the actual test runner that is running the tests.

**Comment by lann** [comment]
Yeah I figured it would be emitted by separate bits of code. Ideally it would all be grouped together but as long as it all ends up in output for debugging purposes I'd be happy
